### PR TITLE
create encounter - initial commit

### PR DESCRIPTION
**Pull request by sashrika**
https://issues.openmrs.org/browse/FM-128 

Few other use cases needs to addressed

### CODE REVIEW

**Comment by harsha89** [comment]
we will need testcases too.

**Comment by sashrika** [comment]
@harsha89 , i will add them once.

File: api/src/main/java/org/openmrs/module/fhir/api/impl/EncounterServiceImpl.java [hunk 1]
```diff
@@ -288,4 +293,57 @@ public void deleteEncounter(String id) {
...
+	@Override
+	public Encounter createFHIREncounter(Encounter encounter) {
+		List<String> errors = new ArrayList<String>();
+		org.openmrs.Encounter encounterToCreate = null;
+		ResourceReferenceDt encounterref = encounter.getPartOf();
+		Visit visit = null;
+		if (encounterref != null && !encounterref.isEmpty()) { // if partOf is not empty, This Encounter should be created under an Visit
+			encounterToCreate = FHIREncounterUtil.generateOMRSEncounter(encounter, errors);
+			IdDt ref = encounterref.getReference();
+			String encounterrefuuid = ref.getIdPart();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/EncounterServiceImpl.java]: use camel case encounterRefuuid

File: api/src/main/java/org/openmrs/module/fhir/api/impl/EncounterServiceImpl.java [hunk 2]
```diff
@@ -288,4 +293,57 @@ public void deleteEncounter(String id) {
...
 	}
+	
+	/**
+	 * @see org.openmrs.module.fhir.api.EncounterService#createFHIREncounter(Encounter)
+	 */
+	@Override
+	public Encounter createFHIREncounter(Encounter encounter) {
+		List<String> errors = new ArrayList<String>();
+		org.openmrs.Encounter encounterToCreate = null;
+		ResourceReferenceDt encounterref = encounter.getPartOf();
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/EncounterServiceImpl.java]: use camel case

Comment (sashrika) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/EncounterServiceImpl.java]: @harsha89  I will fix it.

**Pull request merged**